meshmode/mesh/processing.py

```diff
@@ -108,28 +108,33 @@ def _filter_mesh_groups(groups, selected_elements, vertex_id_dtype):
     group_elem_starts = [np.searchsorted(selected_elements, grp.element_nr_base)
                 for grp in groups] + [len(selected_elements)]
 
-    n_new_groups = 0
-    group_to_new_group = [None] * len(groups)
+    new_group_to_old_group = []
     filtered_group_elements = []
     for igrp, grp in enumerate(groups):
         start_idx, end_idx = group_elem_starts[igrp:igrp+2]
         if end_idx == start_idx:
             continue
 
-        group_to_new_group[igrp] = n_new_groups
+        new_group_to_old_group.append(igrp)
         filtered_group_elements.append(selected_elements[start_idx:end_idx]
                     - grp.element_nr_base)
-        n_new_groups += 1
+
+    n_new_groups = len(new_group_to_old_group)
+
+    group_to_new_group = [None] * len(groups)
+    for i_new_grp, i_old_grp in enumerate(new_group_to_old_group):
+        group_to_new_group[i_old_grp] = i_new_grp
+
+    del grp
```

Comment:
What does this line do?

Reply:
It deletes the local variable `grp`, preventing unintended use of the variable below. (Like a less clean version of making a sub-scope `{}` in C.) I hit a bug where I inadvertently used `grp` without re-setting it, and so I decided to get rid of it up front to avoid that risk.